src/mlpack/methods/ann/layer/celu.hpp

```diff
@@ -0,0 +1,203 @@
+/**
+ * @file celu.hpp
+ * @author Gaurav Singh
+ *
+ * Definition of the CELU activation function as descibed by Jonathan T. Barron.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CELU_HPP
+#define MLPACK_METHODS_ANN_LAYER_CELU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The CELU activation function, defined by
```

Comment:
Hi @gaurav-singh1998, could you place the definition and reference to the paper in the first comment as other activation functions have done the same and split @ tparam part, I think it will look a bit cleaner.

Reply:
The CELU function has been defined mathematically and the reference to the paper also has been given. Do I also need to define it separately?

Reply:
I meant placing definition and research paper in line 6 and keeping only comment about template here.

Reply:
Refer mish activation function, reference to paper and definition at top, above class only @t param and that this class defines CELU activation function.

Reply:
Thanks, I will refer to it.

Reply:
I think what @kartikdutt18 meant was to put:

```
 * @code
 * @article{
 *   author  = {Jonathan T. Barron},
 *   title   = {Continuously Differentiable Exponential Linear Units},
 *   year    = {2017}
 * }
 * @endcode
```

here, right before the function definition.

Reply:
Hi @zoq, In most of the activation function reference was provided at the top and some layers have reference just before function definition. I thought this looks much cleaner else we end up having a very large comment block. 
What do you think?
In functions such as softplus, softsign, etc. the paper has been referenced in the first comment (just above BSD-3-Clause)
Thanks.